src/v/rpc/transport.cc

```diff
@@ -219,22 +219,31 @@ void transport::dispatch_send() {
                          || _requests_queue.begin()->first
                               > (_last_seq + sequence_t(1));
               },
-              [this] {
+              [this]() -> ss::future<> {
                   auto it = _requests_queue.begin();
                   _last_seq = it->first;
-                  auto buffer = std::move(it->second->buffer).get();
-                  // These units are released once we are out of scope here
+                  auto entry = std::move(it->second);
+                  _requests_queue.erase(it->first);
+                
+                  // `entry->resource_units` units are released once we are out of scope here
                   // and that is intentional because the underlying write call
                   // to the batched output stream guarantees us the in-order
                   // delivery of the dispatched write calls, which is the intent
                   // of holding on to the units up until this point.
-                  auto units = std::move(it->second->resource_units);
-                  auto v = std::move(*buffer).as_scattered();
-                  auto msg_size = v.size();
-                  _requests_queue.erase(it->first);
-                  auto f = _out.write(std::move(v));
-                  return std::move(f).finally(
-                    [this, msg_size] { _probe.add_bytes_sent(msg_size); });
+
+                  return ss::do_with(
```

Comment:
Non-obvious what the intent of this change is, and as Noah points out the commit message says something unrelated.